lib/bit_array.h

```diff
@@ -79,12 +79,21 @@ uint8_t bit_array_check_byte( bit_array * this, size_t index );
 int bit_array_add_bit( bit_array * this, bool bit_value );
 
 /**
+ * NOTE: This adds byte starting at MSB.
  * Adds 8 bits to the array specified by bit_value param. This will increase
  * the bit_length of the array by 8. If new bit_length of the array exceeds bit_capacity,
  * the capacity will be doubled to make room.
  */
 int bit_array_add_byte( bit_array * this, uint8_t byte_value );
 
+/**
+ * NOTE: This adds byte starting at LSB.
```

Comment:
I wonder if this is descriptive enough. I'll ask some our our colleagues to ensure they understand the reasoning.